src/components/Select/Select.tsx

```diff
@@ -166,7 +166,8 @@ export const Select = React.forwardRef<HTMLButtonElement, SelectProps>(function
         validationState === 'invalid' && Boolean(errorMessage) && errorPlacement === 'outside';
     const isErrorIconVisible =
         validationState === 'invalid' && Boolean(errorMessage) && errorPlacement === 'inside';
-    const isErrorStateVisible = isErrorMsgVisible || isErrorIconVisible;
+    const isErrorStateVisible =
+        validationState === 'invalid' || isErrorMsgVisible || isErrorIconVisible;
```

Comment:
```suggestion
    const isErrorStateVisible = validationState === 'invalid';
```
And maybe use `isErrorStateVisible` in previous checks

Reply:
done